src/core.jl

```diff
@@ -0,0 +1,138 @@
+import Base: push!, eltype, close
+export Signal, Input, Node, push!, value, close
+
+##### Node #####
+
+immutable Action
+    recipient::WeakRef
+    f::Function
+end
+isrequired(a::Action) = a.recipient.value != nothing && a.recipient.value.alive
+
+type Node{T}
+    value::T
+    parents::Tuple
+    actions::Vector{Action}
+    alive::Bool
+end
+Node(x, parents=()) = Node(x, parents, Action[], true)
+Node{T}(::Type{T}, x, parents=()) = Node{T}(x, parents, Action[], true)
```

Comment:
Would it be possible to switch this to:
```
Base.call{T}(::Type{Node{T}}, x, parents=()) =  Node{T}(x,parents,Action[], true)
```
So this can be backwards compatible on at least 0.4? e.g. `Input{Int64}(1)` will still work.